[FEAT] 강의 단일 조회 API 구현

JSeHoone opened the pull request:
<!-- PR 제목 : [Commit Type] PR_내용 -->
<!-- PR 내용의 경우, 이슈 제목을 그대로 써도 되고, 이슈에 언급되지 않은 내용까지 써주세용 -->
<!-- ex) [FEAT] 회원 API 구현 -->

## ❗️ 이슈 번호
#75 

## 📝 작업 내용
1. 단일 Lecture의 정보를 반환합니다.
2. filePath를 반환 정보에 추가합니다.

```json
{
    "status": 200,
    "message": "OK",
    "data": {
        "lectureId": 1,
        "title": "스프링 프레임워크란?",
        "filePath": "intro.mp4",
        "lectureType": "VIDEO",
        "orderNo": 1
    },
    "code": "OK"
}
```

## 💭 주의 사항

## 💡 리뷰 포인트
<!-- PR 제목 : [Commit Type] PR_내용 -->
<!-- PR 내용의 경우, 이슈 제목을 그대로 써도 되고, 이슈에 언급되지 않은 내용까지 써주세용 -->
<!-- ex) [FEAT] 회원 API 구현 -->

JSeHoone committed:
feat : LectureDetail 관련 DTO 파일 추가

JSeHoone committed:
feat : Lecture Detail API 기능 추가

dnya0 commented on src/main/java/com/example/projectlxp/lecture/service/impl/LectureServiceImpl.java:
```diff
@@ -141,6 +143,29 @@ public void removeLecture(LectureDeleteDTO deleteInfo) {
+        // find Lecture
+        Lecture findLecture = findLectureAndException(lectureId);
+
+        // validate lecture authority
+        lectureValidator.validateLectureAuthority(
+                findLecture.getSection().getCourse().getInstructor().getId(), userId);
```
이거 n+1 문제 없는거 맞지요..?

JSeHoone replied:
확인해보겠습니당 !

JSeHoone replied:
@dnya0 
해당 코드로 Select 쿼리문이 3번 발생하고 있었습니다 !
GPT에게 물어보니 현재는 단 하나의 Lecture에 대한 상세 조회여서 N+1문제는 발생하지 않았다고 했습니다. 

하지만 N+1 문제가 발생할 수 있어서 Fetch Join 쿼리문으로 수정해서 push 했습니다 !

JSeHoone committed:
refactor : findDetailById 쿼리로 수정

- 기존 Lecture에서 section과 course를 가져와서 instructor id를 추출해왔습니다.
- 이 부분에서 N+1 문제가 발생할 수 있기 때문에 Fetch Join 쿼리문을 사용하여 1번의 쿼리만 생기도록 수정하였습니다
